src/components/SearchProduct.tsx

```diff
@@ -0,0 +1,37 @@
+import { useState } from "react";
+
+type SearchProductProps = {
+  placeholder: string;
+  onSearch: (value: string) => void;
+};
+
+export function SearchProduct({ placeholder, onSearch }: SearchProductProps) {
+  const [search, setSearch] = useState("");
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const value = event.target.value;
+    setSearch(value);
+    onSearch(value.trim());
+  };
+
+  return (
+    <div className="mx-2">
+      <label htmlFor="input" className="h6">
+        Name:
+      </label>
+      <input
+        type="text"
+        placeholder={placeholder}
+        value={search}
+        onChange={handleChange}
+        style={{
```

Comment:
use consistent styling if we were using tailwind then use tailwind all the other places?